### PR TITLE
add source_map_ids to error span attributes

### PR DESCRIPTION
**Pull request by Abinet18**
# Description

retrieve source map ids of files in stacktrace
add source_map_ids to error span attributes

## Type of change

Delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- Documentation update

# How has this been tested?

Delete options that are not relevant.

- Manual testing
- Added unit tests
- Added integration tests

### CODE REVIEW

**Comment by Abinet18** [comment]
I have read the CLA Document and I hereby sign the CLA

File: packages/web/integration-tests/tests/errors/views/minified_file_errors.ejs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
```
Comment (t2t2) [inline packages/web/integration-tests/tests/errors/views/minified_file_errors.ejs:1]: Actual integration test (spec.js) missing

File: packages/web/src/SplunkErrorInstrumentation.ts [hunk 2]
```diff
@@ -38,9 +38,33 @@ function stringifyValue(value: unknown) {
...
+    //get sourcemap ids and add to span as error.soruce_map_ids
     span.setAttribute('error.stack', limitLen(err.stack.toString(), STACK_LIMIT));
+    const sourcemapIds = parseErrorStack(err.stack);
+    if (sourcemapIds) {
+      span.setAttribute('error.source_map_ids', sourcemapIds);
+    }
+  }
+}
+
+function parseErrorStack(stack: string): string {
```
Comment (t2t2) [inline packages/web/src/SplunkErrorInstrumentation.ts:52]: `npm run lint:fix`

File: packages/web/integration-tests/tests/errors/views/minified_file_errors.ejs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+	<meta charset="UTF-8">
+	<title>Errors test</title>
+  <%- renderAgent({}) %>
+</head>
+<body>
+    <h1>Minified files with source map ids</h1>
+    <button id="button1">Button 1</button>
+    <button id="button2">Button 2</button>
+    <script src="/integration-tests/tests/errors/views/script1.min.js"></script>
+    <script src="/integration-tests/tests/errors/views/script2.min.js"></script>
```
Comment (t2t2) [inline packages/web/integration-tests/tests/errors/views/minified_file_errors.ejs:13]: Should also add a test for `<script type="module">` as [it doesn't set document.currentScript](https://developer.mozilla.org/en-US/docs/Web/API/Document/currentScript) and the URL should be gotten from [`import.meta.url` there](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import.meta)

File: packages/web/integration-tests/tests/errors/views/script1.min.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+document.getElementById("button1").addEventListener("click",function(){throw new Error("Error from script1.js")});
+(function() {
```
Comment (t2t2) [inline packages/web/integration-tests/tests/errors/views/script1.min.js:2]: Probably should also wrap the content here in try catch block (see issue above about document.currentScript not being set in modules, likely could also end up as an issue if someone does something like `jQuery.getScript`)

**Review by t2t2 — APPROVED** [review]

**Pull request merged**